src/overlays/ovl_i4/fox_fo.c

```diff
@@ -1,29 +1,1513 @@
+
+            case 7:
+                Matrix_MultVec3f(gCalcMatrix, &vec, &actor->vwork[4]);
+                Matrix_GetYRPAngles(gCalcMatrix, &actor->vwork[10]);
+                break;
+
+            case 8:
+                Matrix_MultVec3f(gCalcMatrix, &vec, &actor->vwork[5]);
+                Matrix_GetYRPAngles(gCalcMatrix, &actor->vwork[11]);
+                break;
+        }
+    }
+}
+
+s32 func_i4_80188F08(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* ptr) {
+    Actor* actor = (Actor*) ptr;
+
+    gSPSetGeometryMode(gMasterDisp++, G_CULL_BACK);
+
+    if ((actor->state != 0) && (arg0 != 8)) {
+        *arg1 = NULL;
+    }
+
+    switch (arg0) {
+        case 1:
+        case 2:
+            gSPClearGeometryMode(gMasterDisp++, G_CULL_BACK);
+            break;
+
+        case 5:
+            arg3->x += actor->fwork[0];
+            break;
+
+        case 6:
+            arg3->x += actor->fwork[0];
+            break;
+
+        case 7:
+            arg3->x += actor->fwork[0] * 0.7f;
+            break;
+    }
+
+    return 0;
+}
+
+void func_i4_80188FE4(Actor* actor) {
+    Vec3f vec[20];
+
+    Animation_GetFrameData(&D_6007854, 0, vec);
+    Animation_DrawSkeleton(3, &D_6007980, vec, func_i4_80188F08, func_i4_80188DA0, actor, gCalcMatrix);
+
+    if (actor->state == 1) {
+        actor->state = 2;
+    }
+}
+
+void func_i4_8018906C(void) {
+    Actor* actor = &gActors[50];
+
+    Actor_Initialize(actor);
+    actor->obj.status = 1;
+    actor->obj.id = OBJ_ACTOR_195;
+    actor->obj.pos.x = 0.0f;
+    actor->obj.pos.y = 0.0f;
+    actor->obj.pos.z = -9000.0f;
+    actor->unk_0B6 = 11;
+    actor->scale = 0.0f;
+    Object_SetInfo(&actor->info, actor->obj.id);
+}
+
+f32 D_i4_8019EE4C[] = { -200.0f, 200.0f, -50.0f, -2000.0f };
+f32 D_i4_8019EE5C[] = { 0.0f, 30.0f, -90.0f, 0.0f };
+f32 D_i4_8019EE6C[] = { -100.0f, -200.0f, -300.0f };
+s32 D_i4_8019EE78[] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+
+void func_i4_801890EC(Actor* actor, s32 arg1) {
+    Actor_Initialize(actor);
+    actor->obj.status = 1;
+    actor->obj.id = OBJ_ACTOR_195;
+    actor->obj.pos.x = D_i4_8019EE4C[arg1] + gPlayer[0].pos.x;
+    actor->obj.pos.y = D_i4_8019EE5C[arg1] + gPlayer[0].pos.y;
+    actor->obj.pos.z = D_i4_8019EE6C[arg1] + gPlayer[0].unk_138;
+    actor->vel.z = gPlayer[0].unk_0D0;
+
+    Object_SetInfo(&actor->info, actor->obj.id);
+
+    if (arg1 < 3) {
+        actor->iwork[11] = 1;
+        Audio_PlaySfx(0x3100000CU, actor->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+        return;
+    }
+
+    actor->obj.pos.z = -9500.0f;
+    actor->unk_0B6 = 1;
+    actor->vel.z = 22.0f;
+
+    Audio_PlaySfx(0x11030010U, actor->sfxPos, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+    Audio_PlaySfx(0x31024059U, actor->sfxPos, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+}
+
+void func_i4_8018927C(Player* player) {
+    s32 i;
+    Vec3f src;
+    Vec3f dest;
+    Actor* actorPtr = &gActors[0];
+    Actor* actor2 = &gActors[2];
```

Comment:
I think actor2 is unused here. I'm also not sure what the general approach we should have for using the actorPtr like this is. I've been going with
```
Actor* actor0 = &gActors[0];
Actor* actor1 = &gActors[1];
Actor* actor2 = &gActors[2];
Actor* actor3 = &gActors[3];
```
and using these variables in place of actorPtr[x]. everywhere. It makes more sense to be written like this in my head so they could refer to each character but they both should match so idk

Reply:
Yeah, i've been trying to do the same, although it doesn't seem to match in some parts, i'll see what i can do.